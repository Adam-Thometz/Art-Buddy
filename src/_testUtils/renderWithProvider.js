import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;