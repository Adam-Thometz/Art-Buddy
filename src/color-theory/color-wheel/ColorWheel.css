.ColorWheel {
  position: relative;
  border-radius: 50%;
  --radius: -26vh;
  width: 50%;
  height: 65vh;
  margin: 0 4vw;
}

.ColorWheel-color {
  background-color: #D9D9D9;
  border: 3px solid black;
  border-radius: 50%;
  position: absolute;
  top: 26vh;
  left: 14vw;
  height: calc(95vh / 8);
  width: calc(80vw / 11);
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 1.5rem;
  font-weight: 600;
  text-transform: capitalize;
  color: white;
  -webkit-text-stroke-width: 1px;
  -webkit-text-stroke-color: black;
}

.ColorWheel-color:nth-child(1) {
  transform: translateY(var(--radius));
}
.ColorWheel-color:nth-child(2) {
  transform: rotate(30deg) translateY(var(--radius)) rotate(-30deg);
}
.ColorWheel-color:nth-child(3) {
  transform: rotate(60deg) translateY(var(--radius)) rotate(-60deg);
}
.ColorWheel-color:nth-child(4) {
  transform: rotate(90deg) translateY(var(--radius)) rotate(-90deg);
}
.ColorWheel-color:nth-child(5) {
  transform: rotate(120deg) translateY(var(--radius)) rotate(-120deg);
}
.ColorWheel-color:nth-child(6) {
  transform: rotate(150deg) translateY(var(--radius)) rotate(-150deg);
}
.ColorWheel-color:nth-child(7) {
  transform: rotate(180deg) translateY(var(--radius)) rotate(-180deg);
}
.ColorWheel-color:nth-child(8) {
  transform: rotate(-150deg) translateY(var(--radius)) rotate(150deg);
}
.ColorWheel-color:nth-child(9) {
  transform: rotate(-120deg) translateY(var(--radius)) rotate(120deg);
}
.ColorWheel-color:nth-child(10) {
  transform: rotate(-90deg) translateY(var(--radius)) rotate(90deg);
}
.ColorWheel-color:nth-child(11) {
  transform: rotate(-60deg) translateY(var(--radius)) rotate(60deg);
}
.ColorWheel-color:nth-child(12) {
  transform: rotate(-30deg) translateY(var(--radius)) rotate(30deg);
}