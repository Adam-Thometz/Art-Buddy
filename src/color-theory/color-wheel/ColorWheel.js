import React from 'react';

import { useSelector } from 'react-redux';

import './ColorWheel.css';

const ColorWheel = () => {
  const { colorWheel } = useSelector(state => state.colorTheory);

  const colors = Object.keys(colorWheel).map(c => {
    const { color, transform, isShowing } = colorWheel[c];
    const styles = {
      backgroundColor: isShowing.color ? color.hex : '#D9D9D9',
      transform
    };
    return (
      <div className='ColorWheel-color' style={styles}>
        {isShowing.text ? c : null}
      </div>
    );
  });

  return (
    <div className='ColorWheel'>
      {colors}
    </div>
  );
};

export default ColorWheel;