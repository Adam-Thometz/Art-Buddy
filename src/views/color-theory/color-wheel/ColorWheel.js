import { useSelector } from "react-redux";

import "./ColorWheel.css";

const ColorWheel = ({ update = null }) => {
  const { colorWheel } = useSelector((state) => state.colorWheel);

  const colors = Object.keys(colorWheel).map((c) => {
    const { color, isShowing } = colorWheel[c];
    const backgroundColor = isShowing.color ? color.hex : "#D9D9D9";
    const updateFn = update && isShowing.color ? update : null;
    return (
      <div key={c} id={c} className="ColorWheel-color" style={{ backgroundColor }} onClick={updateFn}>
        {isShowing.text ? c : null}
      </div>
    );
  });

  return <div className="ColorWheel">{colors}</div>;
};

export default ColorWheel;
