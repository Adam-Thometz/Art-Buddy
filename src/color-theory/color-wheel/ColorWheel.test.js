import React from "react";

import renderWithProvider from "_testUtils/renderWithProvider";

import ColorWheel from "./ColorWheel";

describe('ColorWheel component', () => {
  it('renders without crashing', () => {
    renderWithProvider(<ColorWheel />);
  });

  it('matches snapshot', () => {
    const { asFragment } = renderWithProvider(<ColorWheel />);
    expect(asFragment()).toMatchSnapshot();
  });
});