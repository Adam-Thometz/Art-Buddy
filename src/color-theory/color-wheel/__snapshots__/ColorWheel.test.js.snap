// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorWheel component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ColorWheel"
  >
    <div
      class="ColorWheel-color"
      style="background-color: rgb(217, 217, 217);"
    />
    <div
      class="ColorWheel-color"
      style="background-color: rgb(217, 217, 217);"
    />
    <div
      class="ColorWheel-color"
      style="background-color: rgb(217, 217, 217);"
    />
    <div
      class="ColorWheel-color"
      style="background-color: rgb(217, 217, 217);"
    />
    <div
      class="ColorWheel-color"
      style="background-color: rgb(217, 217, 217);"
    />
    <div
      class="ColorWheel-color"
      style="background-color: rgb(217, 217, 217);"
    />
    <div
      class="ColorWheel-color"
      style="background-color: rgb(217, 217, 217);"
    />
    <div
      class="ColorWheel-color"
      style="background-color: rgb(217, 217, 217);"
    />
    <div
      class="ColorWheel-color"
      style="background-color: rgb(217, 217, 217);"
    />
    <div
      class="ColorWheel-color"
      style="background-color: rgb(217, 217, 217);"
    />
    <div
      class="ColorWheel-color"
      style="background-color: rgb(217, 217, 217);"
    />
    <div
      class="ColorWheel-color"
      style="background-color: rgb(217, 217, 217);"
    />
  </div>
</DocumentFragment>
`;
